<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowTester</class>
 <widget class="QMainWindow" name="MainWindowTester">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>979</width>
    <height>642</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Тестировщик программ</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <property name="spacing">
     <number>0</number>
    </property>
    <item row="0" column="0" colspan="2">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>2</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>Настройки</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="2" column="0" colspan="2">
         <widget class="QGroupBox" name="groupBox_2">
          <property name="title">
           <string>Текущие версии интерпретаторов</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_5" columnstretch="0,1">
           <item row="0" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Python</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Pascal</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>C/C++</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLabel" name="lblPythonVer">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLabel" name="lblPascalVer">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QLabel" name="lblCver">
             <property name="text">
              <string/>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QPushButton" name="pb_about">
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>О программе</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <spacer name="horizontalSpacer_2">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="0" colspan="2">
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>Поддержка языков программирования</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_4">
           <item row="0" column="2">
            <widget class="QPushButton" name="pbChoosePathPy">
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_5">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>Путь до компилятора Pascal</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="lePythonPath"/>
           </item>
           <item row="2" column="1">
            <widget class="QLineEdit" name="leCpath"/>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Путь до интерпретатора Python</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QLineEdit" name="lePascalPath">
             <property name="enabled">
              <bool>false</bool>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Путь до компилятора C/C++</string>
             </property>
            </widget>
           </item>
           <item row="1" column="2">
            <widget class="QPushButton" name="pbChoosePathPascal">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
           <item row="2" column="2">
            <widget class="QPushButton" name="pbChoosePathC">
             <property name="maximumSize">
              <size>
               <width>30</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>...</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item row="1" column="0">
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Fixed</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>5</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="testing">
       <attribute name="title">
        <string>Проверить задачу</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_6" rowstretch="0,0,0,0,0,1,0" columnstretch="0,0,1,0">
        <property name="spacing">
         <number>0</number>
        </property>
        <item row="2" column="3">
         <widget class="QPushButton" name="pbClearCode">
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Очистить</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_14">
          <property name="text">
           <string>Какая задача проверяется</string>
          </property>
         </widget>
        </item>
        <item row="3" column="3">
         <spacer name="verticalSpacer_2">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Minimum</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>10</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0" rowspan="6">
         <widget class="QListWidget" name="listCurrentTaskTest">
          <property name="toolTip">
           <string>Выделите задачу</string>
          </property>
         </widget>
        </item>
        <item row="6" column="3">
         <widget class="QPushButton" name="pbClearConsole">
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Очистить</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Expanding</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="pbCheckTask">
          <property name="maximumSize">
           <size>
            <width>100</width>
            <height>16777215</height>
           </size>
          </property>
          <property name="text">
           <string>Проверить</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1" colspan="3">
         <widget class="QTextBrowser" name="tbSourceCodeTask">
          <property name="readOnly">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLabel" name="label_10">
          <property name="text">
           <string>Исходный код задачи</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1" colspan="3">
         <widget class="QTextBrowser" name="tbConsole">
          <property name="toolTip">
           <string>В консоли в ответах не выводятся спец. символы</string>
          </property>
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QLabel" name="label_11">
          <property name="text">
           <string>Консоль вывода</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="add_test_cases">
       <attribute name="title">
        <string>Заполнить тесты</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2" rowstretch="0,1,0,0,3,0,0,0,0,1,1,1,0">
        <item row="12" column="2" colspan="2">
         <widget class="QPushButton" name="pbAddCase">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Добавить проверку</string>
          </property>
         </widget>
        </item>
        <item row="8" column="3">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Выходные данные</string>
          </property>
         </widget>
        </item>
        <item row="6" column="3">
         <widget class="QPushButton" name="pbRemoveCase">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="text">
           <string>Удалить  последнюю проверку</string>
          </property>
         </widget>
        </item>
        <item row="8" column="2">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Входные данные</string>
          </property>
         </widget>
        </item>
        <item row="4" column="2" rowspan="2" colspan="2">
         <widget class="QScrollArea" name="scrollArea">
          <property name="widgetResizable">
           <bool>true</bool>
          </property>
          <widget class="QWidget" name="currentTestsScrollArea">
           <property name="geometry">
            <rect>
             <x>0</x>
             <y>0</y>
             <width>693</width>
             <height>161</height>
            </rect>
           </property>
           <layout class="QGridLayout" name="gridLayout_3"/>
          </widget>
         </widget>
        </item>
        <item row="11" column="0">
         <widget class="QPushButton" name="pbAddTask">
          <property name="text">
           <string>Добавить задачу</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" rowspan="10" colspan="2">
         <widget class="QListWidget" name="listTests"/>
        </item>
        <item row="1" column="2" colspan="2">
         <widget class="QTextBrowser" name="tbTextTask">
          <property name="readOnly">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QLabel" name="label_13">
          <property name="text">
           <string>Текст задачи</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="label_12">
          <property name="text">
           <string>Задачи</string>
          </property>
         </widget>
        </item>
        <item row="7" column="3">
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Minimum</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>14</width>
            <height>11</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="11" column="1">
         <widget class="QPushButton" name="pbRemoveTask">
          <property name="text">
           <string>Удалить задачу</string>
          </property>
         </widget>
        </item>
        <item row="9" column="3" rowspan="3">
         <widget class="QTextBrowser" name="data_out">
          <property name="toolTip">
           <string>Необходимо обращать внимание на лишние пробелы или переводы строк</string>
          </property>
          <property name="readOnly">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="12" column="0">
         <widget class="QPushButton" name="pbImport">
          <property name="toolTip">
           <string>Импорт ранее заполненных задач</string>
          </property>
          <property name="text">
           <string>Импорт из файла</string>
          </property>
         </widget>
        </item>
        <item row="9" column="2" rowspan="3">
         <widget class="QTextBrowser" name="data_in">
          <property name="toolTip">
           <string>Необходимо обращать внимание на лишние пробелы или переводы строк</string>
          </property>
          <property name="toolTipDuration">
           <number>-1</number>
          </property>
          <property name="readOnly">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="12" column="1">
         <widget class="QPushButton" name="pbExport">
          <property name="toolTip">
           <string>Экспорт всех заполненных задач в специальный файл, который можно импортиовать в эту программу на другом компьюетре</string>
          </property>
          <property name="text">
           <string>Экспорт в файл</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Minimum</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>14</width>
            <height>11</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="3" column="2" colspan="2">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Проверки</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>979</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
